### DISCUSSION TITLE
Difference between standard and legacy retry modes

### DISCUSSION THREAD

**Discussion by lonix1** [aws/aws-sdk-net#3522] [Q&A] [answered]
I'm using SESv2 but the underlying issue probably applies to multiple SDK classes.

The [RequestRetryMode](https://docs.aws.amazon.com/sdkfornet/v3/apidocs/items/Runtime/TRequestRetryMode.html) enum:
- Adaptive - "An experimental request retry strategy that builds on the Standard strategy and introduces congestion control through client side rate limiting."
- Legacy - "Legacy request retry strategy."
- Standard - "Standardized request retry strategy that is consistent across all SDKs."

What is the difference between legacy and standard? I can find no documentation anywhere which explains when to use one or the other. Does legacy mean v1 of the API?

This is confusing because `AmazonSimpleEmailServiceV2Config.ClientConfig.RetryMode` has a default value of Legacy.

**Comment by normj** [comment, marked as answer]
Sorry for the lack of documentation. The difference between `Legacy` and `Standard` is `Legacy` has a max retry delay of 30 seconds versus 20 seconds for `Standard` and `Standard` adds some jitter in the retry delay. In the upcoming V4 of the SDK the `Legacy` enum is going away.

**Reply by lonix1** [reply]
Thanks. I also found [this info](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-retries.html) but for the CLI. I assume it's the same for the SDK.

**Reply by normj** [reply]
The `Legacy` mode would be different between CLI and .NET. The whole point of creating the `Standard` was to get the SDKs to have the same retry rules and `Legacy` is the old SDK specific implementation.
